types.go

```diff
@@ -83,3 +61,70 @@ func (m PubKey) AsABCI() abci.PubKey {
 		Type: m.Type,
 	}
 }
+
+func (m ValidatorUpdates) Validate() error {
+	var err error
+	for _, v := range m.ValidatorUpdates {
+		err = errors.Append(err, v.Validate())
+	}
+	return err
+}
+
+// Deduplicate makes sure we only use the last validator update to any given validator.
+// For bookkeeping we have an option to drop validators with zero power, because those
+// are being remove by tendermint once propagated.
+func (m ValidatorUpdates) Deduplicate(dropZeroPower bool) []ValidatorUpdate {
```

Comment:
Why not return ValidatorUpdates? Seems simpler to use that way.
I like that it does not modify in place